mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/EventDispatcher.java

```diff
@@ -52,16 +54,29 @@ void start() {
             .block();
     }
 
-    Mono<Void> dispatch(Event event) {
-        Mono<OutboundMessage> outboundMessage = Mono.just(event)
+    Mono<Void> dispatch(Event event, Set<RegistrationKey> keys) {
+        Mono<byte[]> serializedEvent = Mono.just(event)
             .publishOn(Schedulers.parallel())
             .map(this::serializeEvent)
-            .map(payload -> new OutboundMessage(MAILBOX_EVENT_EXCHANGE_NAME, EMPTY_ROUTING_KEY, payload));
+            .cache();
 
-        return sender.send(outboundMessage)
+        return doDispatch(serializedEvent, keys)
             .subscribeWith(MonoProcessor.create());
     }
 
+    private Mono<Void> doDispatch(Mono<byte[]> serializedEvent, Set<RegistrationKey> keys) {
+        Flux<String> routingKeys = Flux.concat(
```

Comment:
Can't we have empty-routing-key being typed `RoutingKey` and change the type to `Flux<RoutingKey>`?